bool IR_1
bool IR_2
bool IR_3
bool IR_4
bool IR_5
---
string state
